torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -790,12 +790,13 @@ void InitXlaModuleBindings(py::module m) {
       m, "XlaShardingSpec")
       .def(py::init([](at::Tensor tensor, const py::list& tile_assignment,
                        const py::list& group_assignment,
-                       const py::list& replication_groups, int sharding_type) {
+                       const py::list& replication_groups, int sharding_type,
+                       bool minibatch) {
         return std::make_shared<XLATensor::ShardingSpec>(
             ShardingUtil::CreateOpSharding(
                 tile_assignment, group_assignment, replication_groups,
                 ShardingUtil::ShardingType(sharding_type)),
-            CreateComputationShapeFromTensor(tensor, nullptr));
+            CreateComputationShapeFromTensor(tensor, nullptr), minibatch);
```

Comment:
We need to attach the global shape to the ShardingSpec here. `CreateComputationShapeFromTensor(tensor, nullptr)` will return the shape of the minibatch.

I think we get the correct xla::Shape by using `CreateComputationShapeFromTensor` on the minibatch to get the local shape and use `set_dimensions` to set dim 0 to `tensor.sizes[0] * global_device_count / local_device_count`.

We'll also need to enforce that all devices in the tiling assignment are on the batch axis.